adafruit_vcnl4020.py

```diff
@@ -16,22 +20,265 @@
+    """Reads the ambient light/lux sensor (ALS) measurement result"""
+    _lux_averaging = RWBits(3, _REG_AMBIENT_PARAM, 0)
+    lux_enabled = RWBit(_REG_COMMAND, 2)
+    """Enable/disable lux sensor"""
+    lux_on_demand = RWBit(_REG_COMMAND, 4)
+    """On-demand setting for lux measurements"""
+    _lux_rate = RWBits(3, _REG_AMBIENT_PARAM, 4)
+    lux_ready = ROBit(_REG_COMMAND, 6)
+    """Status of ambient light data"""
+    proximity = ROUnaryStruct(_REG_PROX_RESULT_HIGH, ">H")
+    """Reads the proximity measurement result"""
+    proximity_enabled = RWBit(_REG_COMMAND, 1)
+    """Enable/disable proximity sensor"""
+    _proximity_frequency = RWBits(2, _REG_PROX_ADJUST, 3)
+    promixity_on_demand = RWBit(_REG_COMMAND, 3)
+    """On-demand setting for proximity measurements"""
+    _proximity_rate = RWBits(3, _REG_PROX_RATE, 0)
+    proximity_ready = ROBit(_REG_COMMAND, 5)
+    """Status of proximity data."""
+    low_threshold = Struct(_REG_LOW_THRES_HIGH, ">H")
+    """Sets the low threshold for proximity measurement"""
+    high_threshold = Struct(_REG_HIGH_THRES_HIGH, ">H")
+    """Sets the high threshold for proximity measurement."""
+    _interrupt_count = RWBits(3, _REG_INT_CTRL, 5)
+    proximity_interrupt = RWBit(_REG_INT_CTRL, 3)
+    """Enable/disable proximity interrupt"""
+    lux_interrupt = RWBit(_REG_INT_CTRL, 2)
+    """Enable/disable lux interrupt"""
+    high_threshold_interrupt = RWBit(_REG_INT_CTRL, 1)
+    """Enable/disable proximity high threshold interrupt"""
+    low_threshold_interrupt = RWBit(_REG_INT_CTRL, 0)
+    """Enable/disable proximity low threshold interrupt"""
+    selftimed_enabled = RWBit(_REG_COMMAND, 0)
+    """Enable/disable selftimed reading"""
+
+    _proximity_rates = [1.95, 3.9, 7.8, 16.6, 31.2, 62.5, 125, 250]
+    _lux_rates = [1, 2, 3, 4, 5, 6, 8, 10]
+    _avg_samples = [1, 2, 4, 8, 16, 32, 64, 128]
+    _int_counts = [1, 2, 4, 8, 16, 32, 64, 128]
+    _proximity_frequencies = [390.625, 781.25, 1.5625, 3.125]
+
+    def __init__(self, i2c: I2C, addr: int = _I2C_ADDRESS) -> None:
+        """
+        Initializes the VCNL4020 sensor and checks for a valid Product ID Revision.
+        :param i2c: The I2C interface to use
+        :param addr: The I2C address of the VCNL4020, defaults to _I2C_ADDRESS
+        """
+        self.i2c_device = I2CDevice(i2c, addr)
+
+        # Check the Product ID Revision
+        if self._product_revision != 0x21:
+            raise RuntimeError(f"Invalid Product ID Revision {self._product_revision}")
+        try:
+            # Configuration settings
+            self.proximity_rate = 250
+            self.led_current = 200
+            self.lux_rate = 10
+            self.lux_averaging = 1
+        except Exception as error:
+            raise RuntimeError(f"Failed to initialize: {error}") from error
+
+    @property
+    def _enable(self) -> bool:
+        return self.lux_enabled, self.proximity_enabled, self.selftimed_enabled
```

Comment:
This doesn't return a bool. It'll be a tuple of three bools. I think you want to `and` the three values together.